torch/_dynamo/variables/lists.py

```diff
@@ -624,7 +624,14 @@ def python_type(self):
         return slice
 
     def as_python_constant(self):
-        return slice(*[x.as_python_constant() for x in self.items])
+        def _to_constant(arg):
+            if isinstance(arg, variables.SymNodeVariable):
+                return arg.evaluate_expr()
+            else:
+                return arg.as_python_constant()
```

Comment:
This is a copy of 

```
def guard_if_dyn(arg):
    if isinstance(arg, SymNodeVariable):
        return arg.evaluate_expr(tx.output)
    elif isinstance(arg, ConstantVariable):
        return arg.as_python_constant()
    return arg
```

and

```
            if isinstance(dim_var, SymNodeVariable):
                # This is because SymNodeVariable intentionally doesn't define
                # as_python_constant to avoid shunting down some codepaths
                # that expect consts.   In this case, we know we definitely
                # want to specialize though.
                dim = dim_var.evaluate_expr()
            elif dim_var is not None:
                dim = dim_var.as_python_constant()
```

No more copying please.
Move it into a util, and refactor all the callsites.

If you want to do that in a PR before this one, fine by me!

Reply:
Refactored in this PR.